graphql/distro_resolver.go

```diff
@@ -25,21 +28,50 @@ func (r *distroResolver) ProviderSettingsList(ctx context.Context, obj *model.AP
 func (r *distroInputResolver) ProviderSettingsList(ctx context.Context, obj *model.APIDistro, data []map[string]interface{}) error {
 	settings := []*birch.Document{}
 	for _, entry := range data {
-		doc, err := birch.DC.MapInterfaceErr(entry)
+		newEntry, err := json.Marshal(entry)
 		if err != nil {
-			return InternalServerError.Send(ctx, "converting provider settings list to birch documents")
+			return InternalServerError.Send(ctx, fmt.Sprintf("marshalling provider settings entry: %s", err.Error()))
+		}
+		doc := &birch.Document{}
+		err = json.Unmarshal(newEntry, doc)
+		if err != nil {
+			return InternalServerError.Send(ctx, fmt.Sprintf("converting map to birch: %s", err.Error()))
 		}
 		settings = append(settings, doc)
 	}
 	obj.ProviderSettingsList = settings
 	return nil
 }
 
+// AcceptableHostIdleTime is the resolver for the acceptableHostIdleTime field.
+func (r *hostAllocatorSettingsInputResolver) AcceptableHostIdleTime(ctx context.Context, obj *model.APIHostAllocatorSettings, data int) error {
+	obj.AcceptableHostIdleTime = model.NewAPIDuration(time.Duration(obj.AcceptableHostIdleTime))
+	return nil
+}
```

Comment:
This confuses me `AcceptableHostIdleTime` is already an `APIDuration` why is it being recast into `Duration` and then an `APIDuration` again?

Reply:
This is a resolver for the input field specifically, which is an `Int`. I ran into trouble passing a `Duration` object to the resolver as an input field; we don't do that anywhere else in the codebase and I'm not sure what it would look like. If you know how that could be done we can avoid this casting from `Int` to `APIDuration`!

Reply:
hmm I'm also confused because like Mohamed said, `AcceptableHostIdleTime` should already be an `APIDuration`. Are we not supposed to be using the `data` field (which is an `int`) instead here?

Reply:
I think there may be a typo here it doesn't look like you are doing anything with an int. Did you mean to pass in data?